### DISCUSSION TITLE
apm generate mxmlc-config

### DISCUSSION THREAD

**Discussion by j3k0** [airsdk/apm#58] [Ideas]
To compile the application's SWF file, we're using `mxmlc`, with an `mxmlc-config.xml` file loaded using `-load-config mxmlc-config.xml`.

The list of ANEs need to be included in this file under `<compiler><library-path>...`.

I wrote a shell script for this but I thought maybe a better place to have this functionality would be in the `apm` tool itself. What about an option to generate a full mxmlc-config file from a template (same process as with the application-descriptor). A default template could be provided by `apm` to make jumping into a new project easier.

Below is our script, which is a little specific to how our project is organized, but just so you get the idea.

```bash
#!/bin/bash
PLATFORM_TYPE="$1"
if [ "x$1" = x ]; then
    echo "usage: $0 desktop|mobile"
    exit 1
fi

if test -e "$(pwd)/build/apm-$PLATFORM_TYPE/ane"; then
    ANES="$(find "$(pwd)/build/apm-$PLATFORM_TYPE/ane" -name '*.ane')"
else
    ANES=
fi

echo '<flex-config>'
echo ' <compiler>'
echo '  <library-path>'
for ANE in $ANES; do
    echo "   <path-element>$ANE</path-element>"
done
echo '  </library-path>'
echo ' </compiler>'
echo '</flex-config>'
```

**Comment by marchbold** [comment]
Yeah, definitely one of the endpoints for this tool is generating build scripts. 

Trying not to over complicate the v1 release though, so was thinking things like this might go into the next release.
